### DISCUSSION TITLE
How to develop components that depend on scripts being in the <head>?

### DISCUSSION THREAD

**Discussion by AlexEBall** [teambit/bit#4807] [Help]
Originally posted on #technical-questions Slack channel 

if developing a map component that needs google maps script in the head, or any component as a matter of fact that requires some script in the head of the html document, does Bit give us a way to do that?

I tried to load the script to the head dynamically using DOM manipulation upon mount of the component but it mounts it to the iframe head and not the top level document head which I believe in this case where google maps api is searching for it and why it errors

cc: @itaymendel

**Comment by itaymendel** [comment]
Thanks for opening a discussion thread on the topic.

First of all, depending on your framework there are several ways to achieve managing <head> and loading third party javascript bundles. Which framework you are using for building your component? React/Angular/VanillaJS/?...

**Reply by AlexEBall** [reply]
My bad. Should have specified that in the original question. We are using React.

**Comment by KutnerUri** [comment]
User code (including preview and docs), is isolated in a separate iframe.
You cannot edit the window of the main site for security reasons (otherwise malicious users could use your cookie and make api calls)

![Screen Shot 2021-08-30 at 11 24 16](https://user-images.githubusercontent.com/5400361/131310413-1e2ee699-93e6-48d1-a7d2-c5135a3a0b74.png)

I don't understand your use case. If you add a script to the head tag, it should execute. It doesn't matter which window it's attached to. Maybe open the composition in a new page?
Are you using some browser extension, that only looks at the top level window?
